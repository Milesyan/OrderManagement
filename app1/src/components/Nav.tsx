import React from 'react';
import styles from './Nav.module.css';

export default function Nav() {
  return (
    <div className={styles.navRoot}>
      <div className={styles.container}>
        Food Orders Manage System
      </div>
    </div>
  )
}