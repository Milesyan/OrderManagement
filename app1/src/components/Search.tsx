import {useEffect, useState} from "react"
import {useDebounce} from "../common/hooks"
import styles from './Search.module.css';

interface ISearch {
  onSearchTermUpdate: (t: string) => void;
  count: number;
}
export default function Search(props: ISearch) {
  const [searchTerm, setSearchTerm] = useState('');
  const onInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    const rawValue = e.target.value;
    const value = rawValue.split(".").slice(0, 2).map((ele, i) => i ? ele.slice(0, 2) : ele).join(".")
    setSearchTerm(value);
  }
  const debouncedValue = useDebounce(searchTerm, 200)
  useEffect(() => {
    props.onSearchTermUpdate(debouncedValue);
  }, [debouncedValue])
  return (
    <div className={styles.searchRoot}>
      <input
        className={styles.input}
        id='search-input'
        value={searchTerm}
        type="text"
        defaultValue=""
        placeholder="Search by price (e.g. 34.56)"
        onChange={onInputChange}
        autoComplete={"off"}
      />
      {
        debouncedValue &&
        <div className={styles.count}>
          Matching Orders: {props.count}
        </div>
      }

    </div>
  )
}