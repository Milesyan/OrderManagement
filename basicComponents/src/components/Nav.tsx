import React from 'react';
import styles from './Nav.module.css';

interface INav {
  title: string;
}
export default function Nav(props: INav) {
  return (
    <div className={styles.navRoot}>
      <div className={styles.container}>
        {props.title}
      </div>
    </div>
  )
}