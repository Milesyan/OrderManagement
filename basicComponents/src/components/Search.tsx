import {useEffect, useState} from "react"
import {useDebounce} from "../common/hooks"
import styles from './Search.module.css';

interface ISearch {
  onSearchTermUpdate: (t: string) => void;
  text: string;
  placeholder?: string;
}
export default function Search(props: ISearch) {
  const [searchTerm, setSearchTerm] = useState('');
  const onInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    const rawValue = e.target.value;
    const value = rawValue.split(".").slice(0, 2).map((ele, i) => i ? ele.slice(0, 2) : ele).join(".")
    setSearchTerm(value);
  }
  const debouncedValue = useDebounce(searchTerm, 200)
  useEffect(() => {
    props.onSearchTermUpdate(debouncedValue);
  }, [debouncedValue])
  return (
    <div className={styles.searchRoot}>
      <input
        className={styles.input}
        value={searchTerm}
        type="number"
        step={0.01}
        placeholder={props.placeholder}
        onChange={onInputChange}
        autoComplete={"off"}
      />
      {
        debouncedValue &&
        <div className={styles.count}>
          {props.text}
        </div>
      }

    </div>
  )
}