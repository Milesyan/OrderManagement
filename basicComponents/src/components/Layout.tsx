import React, {PropsWithChildren} from 'react';
import styles from './Layout.module.css';
import Nav from './Nav';

interface ILayout {

}
export default function Layout(props: PropsWithChildren<ILayout>) {
  return (
    <>
      <div className={styles.root}>
        <Nav/>
        {props.children}
      </div>
    </>
  )
}