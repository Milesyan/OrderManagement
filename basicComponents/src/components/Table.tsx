import {ReactNode} from 'react';
import Divider from '../common/Divider';
import Search from './Search';
import styles from './Table.module.css';

interface ITable {
  text?: string;
  supportSearch?: boolean;
  searchPlaceholder?: string;
  renderHeader?: () => ReactNode;
  renderContent?: () => ReactNode;
  onSearchUpdate?: (term: string) => void;
}
export default function Table(props: ITable) {
  return (
    <>
      {
        props.supportSearch && props.onSearchUpdate &&
        <>
          <Search
            placeholder={props.searchPlaceholder}
            onSearchTermUpdate={props.onSearchUpdate} text={props.text ?? ''} />
          <Divider />
        </>
      }
      <table className={styles.table}>
        <thead>
          {props.renderHeader?.()}
        </thead>
        <tbody>
          {props.renderContent?.()}
        </tbody>
      </table>
    </>
  );
}

