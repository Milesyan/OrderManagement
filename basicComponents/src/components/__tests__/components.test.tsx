import Layout from '../Layout';
import Nav from '../Nav';
import Search from '../Search';
import Table from '../Table';
import TestRenderer from 'react-test-renderer';

test('render Layout components', () => {
  const ele = TestRenderer.create(<Layout title="Demo title"/>)
  expect(ele).toMatchSnapshot()
});

test('render Nav components', () => {
  const ele = TestRenderer.create(<Nav title="Demo Nav title"/>)
  expect(ele).toMatchSnapshot()
});

test('render Search components', () => {
  const ele = TestRenderer.create(<Search text="demo Search" onSearchTermUpdate={() => {}}/>)
  expect(ele).toMatchSnapshot()
});

test('render Table components', () => {
  const ele = TestRenderer.create(<Table 
      renderHeader={() => <div>Example header</div>}
      renderContent={() => <div>Exmaple content</div>}
  />)
  expect(ele).toMatchSnapshot()
});
