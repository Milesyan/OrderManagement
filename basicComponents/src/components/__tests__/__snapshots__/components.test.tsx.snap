// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Layout components 1`] = `
<div
  className="root"
>
  <div
    className="navRoot"
  >
    <div
      className="container"
    >
      Demo title
    </div>
  </div>
</div>
`;

exports[`render Nav components 1`] = `
<div
  className="navRoot"
>
  <div
    className="container"
  >
    Demo Nav title
  </div>
</div>
`;

exports[`render Search components 1`] = `
<div
  className="searchRoot"
>
  <input
    autoComplete="off"
    className="input"
    onChange={[Function]}
    step={0.01}
    type="number"
    value=""
  />
  
</div>
`;

exports[`render Table components 1`] = `
<table
  className="table"
>
  <thead>
    <div>
      Example header
    </div>
  </thead>
  <tbody>
    <div>
      Exmaple content
    </div>
  </tbody>
</table>
`;
